oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Checkpoints.java

```diff
@@ -146,8 +146,13 @@ public Revision getOldestRevisionToKeep() {
         }
 
         if (op.hasChanges()) {
-            store.findAndUpdate(Collection.SETTINGS, op);
-            LOG.debug("Purged {} expired checkpoints", op.getChanges().size());
+            try {
+                store.findAndUpdate(Collection.SETTINGS, op);
+                LOG.debug("Purged {} expired checkpoints", op.getChanges().size());
+            } catch (UnsupportedOperationException uoe) {
+                LOG.info("getOldestRevisionToKeep : could not clean up expired checkpoints"
```

Comment:
It should be `LOG.warn` at least.

Reply:
Reason for info was that using this in read-only (eg via oak-run revisions info), this exception is likely met. In which case a warn would be confusing. I think we'd need to expose whether or not a DocumentStore is read-only somehow - without that we're left with catching this exception (which I agree is not nice)